pepper-apis/src/main/java/org/broadinstitute/ddp/housekeeping/schedule/CheckAgeUpJob.java

```diff
@@ -98,7 +98,8 @@ private void run(Handle handle) {
             }
             try {
                 LOG.info("Running age-up check for study {}", policy.getStudyGuid());
-                service.runAgeUpCheck(handle, interpreter, policy);
+                // Perform check for each study within a transaction so any unhandled errors for study will not affect other studies.
+                TransactionWrapper.useTxn(TransactionWrapper.DB.APIS, handle -> service.runAgeUpCheck(handle, interpreter, policy));
```

Comment:
Since `runAgeUpCheck` runs in its own transaction now, I think it might be cleaner to have `runAgeUpCheck` use three distinct transactions, rather than savepoint/rollback, since explicit savepoints and rollbacks are a bit more complicated.

Reply:
Not sure I fully understand the suggestion. `runAgeUpCheck` is running within a transaction already. Are we suggesting to open additional transactions within this to do the different pieces of change? I was thinking having transactions-within-transactions is discouraged?

If the concern is around managing the savepoints, which I'm also concerned about, I think we can provide methods to help manage the lifecycle for it. Idea here is very similar to what we do for opening a transaction. E.g. let's say we have this method signature in `TransactionWrapper`:
```java
public static <X extends Exception> void useSavepoint(String name, Handle handle, HandleConsumer<X> callback) throws X {}
```
Then we can use it to wrap the "critical region" of the code that mutates data:
```java
try {
    EventSignal signal = new EventSignal(..., REACHED_AOM_PREP);
    TransactionWrapper.useSavepoint(checkpoint("prep", study, user), handle,
            h -> EventService.getInstance().processAllActionsForEventSignal(h, signal));
} catch (Exception e) {
    LOG.error("Error ... rolling back and skipping",);
    continue;
}
```
What do folks think?

Reply:
I like the `useSavepoint()` approach!